pkg/download/downloader.go

```diff
@@ -142,22 +144,45 @@ func extractTARGZ(targetDir string, in io.Reader) error {
 	return nil
 }
 
-func extractArchive(filename, dst string, r io.ReaderAt, size int64) error {
+func detectMIMEType(at io.ReaderAt) (string, error) {
```

Comment:
uh oh I just realized there's no tests for this one :) 

can you add a test with simple strings.NewReader like:

- 0 bytes
- <512 bytes
- >512 bytes
- html content(?)

I have a suspicion that buf you're allocating below has 512 bytes length, so if you're reading only 12 bytes, you have 500 bytes of \0 –which may result in wrong results?

Reply:
mime type matching has some edge cases:
* 0 0-bytes  ->  `text/plain`
* 1+ 0-bytes -> `application/octet-stream`
This should not be a problem for us, as there are no 0 byte zip/targz files.